CONFIGURE.md

```diff
@@ -1,160 +1,243 @@
-* Restart your rails server
-
-An example excerpt from settings.local.yml:
-
-```yaml
-# OAuth 2 Client ID for the web site
-oauth_application: "SGm8QJ6tmoPXEaUPIZzLUmm1iujltYZVWCp9hvGsqXg"
-# OAuth 2 Client Secret for the web site
-oauth_key: "eRHPm4GtEnw9ovB1Iw7EcCLGtUb66bXbAAspv3aJxlI"
+* **iD** - the web-based editor
+* **The website itself** - for the Notes functionality
+
+### Setting up OAuth for iD
+
+1. **Navigate to OAuth applications:**
+   - Go to "[OAuth 2 applications](http://localhost:3000/oauth2/applications)" on the My settings page
+
+2. **Register new application:**
+   - Click on "Register new application"
+   - **Name:** "Local iD"
+   - **Redirect URIs:** "http://localhost:3000" (unless you have set up alternatives)
+
+3. **Select permissions:**
+   Check boxes for the following:
+   - ✅ 'Read user preferences'
+   - ✅ 'Modify user preferences'
+   - ✅ 'Modify the map'
+   - ✅ 'Read private GPS traces'
+   - ✅ 'Upload GPS traces'
+   - ✅ 'Modify notes'
+
+4. **Configure the application:**
+   - Copy the "Client ID" from the next page
+   - Edit `config/settings.local.yml` in your rails tree
+   - Add the "id_application" configuration with the "Client ID" as the value
+   - Restart your rails server
+
+> [!TIP]
+> **Example configuration in `settings.local.yml`:**
+> ```yaml
+> # Default editor
+> default_editor: "id"
+> # OAuth 2 Client ID for iD
+> id_application: "Snv…OA0"
+> ```
+
+### Setting up OAuth for Notes and Changeset Discussions
+
+To allow [Notes](https://wiki.openstreetmap.org/wiki/Notes) and changeset discussions to work:
+
+1. **Register OAuth application for the website:**
+   - Go to "[OAuth 2 applications](http://localhost:3000/oauth2/applications)" on the My settings page
+   - Click on "Register new application"
+   - **Name:** "OpenStreetMap Web Site"
+   - **Redirect URIs:** "http://localhost:3000"
+
+2. **Select permissions:**
+   Check boxes for:
+   - ✅ 'Modify the map'
+   - ✅ 'Modify notes'
+
+3. **Configure the application:**
+   - Copy both the "Client Secret" and "Client ID"
+   - Edit `config/settings.local.yml`
+   - Add both configurations
+   - Restart your rails server
+
+> [!TIP]
+> **Example configuration in `settings.local.yml`:**
+> ```yaml
+> # OAuth 2 Client ID for the web site
+> oauth_application: "SGm8QJ6tmoPXEaUPIZzLUmm1iujltYZVWCp9hvGsqXg"
+> # OAuth 2 Client Secret for the web site
+> oauth_key: "eRHPm4GtEnw9ovB1Iw7EcCLGtUb66bXbAAspv3aJxlI"
+> ```
+
+## Development Tools
+
+### Viewing Rails Logs
+
+Rails has its own log. To inspect the log during development:
+
+```bash
+tail -f log/development.log
 ```
 
-## Troubleshooting
+### Maintaining Your Installation
 
-Rails has its own log.  To inspect the log, do this:
+> [!TIP]
+> If your installation stops working for some reason:
+> 
+> - **Update gem dependencies:** Sometimes gem dependencies change. Go to your `openstreetmap-website` directory and run:
```

Comment:
I'd just say `gems` rather than `gem dependencies` as it's not really about dependencies changing, just that the bundle as a whole has been updated.

We should probably also have a bullet point for running `bundle exec bin/yarn install` to update node modules.